Dockerfile

```diff
@@ -74,13 +74,15 @@ RUN set -ex; \
             python3-dev \
             apache-arrow-dev \
             py3-pip; \
+        # Upgrade pip to 24.3.1 to match the version in the ha timescaledb-docker image.
+        python3 -m pip install --upgrade pip==24.3.1 --break-system-packages; \
         git clone --branch ${PGAI_VERSION} https://github.com/timescale/pgai.git /build/pgai; \
         cd /build/pgai; \
         # note: this is a hack. pyarrow will be built from source, so must be pinned to this arrow version \
-        echo pyarrow==$(pkg-config --modversion arrow) >> ./projects/extension/requirements.txt; \
+        echo pyarrow==$(pkg-config --modversion arrow) >> ./projects/extension/requirements-lock.txt; \
```

Comment:
We moved from having a `requirements.txt` file to `requirements-lock.txt` file with locked dependencies (built by `uv`). 
Even though without this change the `install()` from `build.py` will install this (See [install_old_py_deps()](https://github.com/timescale/pgai/blob/main/projects/extension/build.py#L461), this change is necessary for consistency sake; If tomorrow we stop considering installing old dependencies from `requirements.txt` file, this will fail.